File: apps/java/io/dustin/apps/common/utils/TypeCode.java
package io.dustin.apps.common.utils;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor(access = AccessLevel.PROTECTED)
public enum TypeCode {

    ONE("A"),
    TWO("B"),
    THREE("C"),
    FOUR("D");

    @Getter
    private String code;


}

File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/common/utils/TypeCode.kt
package io.dustin.apps.common.utils

import lombok.AccessLevel

@AllArgsConstructor(access = AccessLevel.PROTECTED)
enum class TypeCode {
    ONE("A"),
    TWO("B"),
    THREE("C"),
    FOUR("D");

    @Getter
    private val code: String? = null
}
